src/main/java/eu/arrowhead/client/library/ArrowheadService.java

```diff
@@ -448,6 +457,133 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
 		return response.getBody();
 	}
 	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Make WS connection with the specified service reachability details.
+	 *
+	 * @param handler WebSocket handler to use for message exchanges
+	 * @param address String value which represents the host where the service is available.
+	 * @param port int value which represents the port where the service is available
+	 * @param serviceUri String value which represents the URI where the service is available.
+	 * @param interfaceName String value which represents the name of the interface used for the communication. Usable interfaces could be received in orchestration response.
+	 * @param token (nullable) String value which represents the token for being authorized at the provider side if necessary. Token could be received in orchestration response per interface type.
+	 * @param queryParams (nullable) String... variable arguments which represent the additional key-value http(s) query parameters if any necessary. E.g.: "k1", "v1", "k2", "v2".
+	 * @return the response received from the provider
+	 *
+	 * @throws InvalidParameterException when service URL can't be assembled.
+	 * @throws AuthException when ssl context or access control related issue happened.
+	 * @throws ArrowheadException when the communication is managed via Gateway Core System and internal server error happened.
+	 * @throws UnavailableServerException when the specified server is not available.
+	 */
+	public WebSocketConnectionManager connnectServiceWS(final WebSocketHandler handler, final String address, final int port, final String serviceUri, final String interfaceName,
```

Comment:
You could handles WS ans WSS within one method (whics is much more user friendly). See in the  comment of next method

Reply:
Fixed